internal/db/arango.go

```diff
@@ -118,14 +140,96 @@ func (c *ArangoClient) QueryAnnotations(ctx context.Context, key string) ([]docu
 
 	var annotations []documents.Annotation
 	for {
-		var doc documents.Annotation
-		_, err := cursor.ReadDocument(ctx, &doc)
+		_, err := cursor.ReadDocument(ctx, &annotations)
 		if driver.IsNoMoreDocuments(err) {
 			break
 		} else if err != nil {
 			return nil, err
 		}
-		annotations = append(annotations, doc)
 	}
+
 	return annotations, nil
 }
+
+func (c *ArangoClient) QueryScoreByLayer(ctx context.Context, key string, layer contracts.LayerType) ([]documents.Score, error) {
+	db, err := c.instance.Database(ctx, c.cfg.DatabaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	var query string
+	switch layer {
+	case contracts.Application:
+		query = `FOR s IN scores FILTER s.dataRef == @key AND s.layer == @layer RETURN [s]`
+	case contracts.CiCd:
+		query = `FOR appScore IN scores FILTER appScore.dataRef == @key 
+				LET cicdScore = (
+					FOR s IN scores FILTER 
+					s.layer == @layer AND s.tag ANY IN appScore.tag 
+					RETURN s 
+				)
+				RETURN cicdScore `
+	case contracts.Os, contracts.Host:
+		query = `FOR a in annotations FILTER a.dataRef == @key LIMIT 1
+				LET scores = (FOR s IN scores FILTER s.layer == @layer AND a.host IN s.tag RETURN s)
+				RETURN scores`
+
+	}
+	bindVars := map[string]interface{}{
+		"key":   key,
+		"layer": layer,
+	}
+	cursor, err := db.Query(ctx, query, bindVars)
+	if err != nil {
+		return nil, err
+	}
+	defer cursor.Close()
+
+	var scores []documents.Score
+	for {
+		_, err := cursor.ReadDocument(ctx, &scores)
+		if driver.IsNoMoreDocuments(err) {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+	}
+
+	return scores, nil
+}
+
+func (c *ArangoClient) FetchHosts(ctx context.Context) ([]string, error) {
+	db, err := c.instance.Database(ctx, c.cfg.DatabaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	query := `FOR a IN annotations LET hosts = (a.host) RETURN DISTINCT hosts`
+	cursor, err := db.Query(ctx, query, nil)
```

Comment:
GIven that Host annotations have the machine id from '/etc/machine-id' as the value for the "Host" field, we should be only selecting hosts from app layer annotations, or exclude host layer annotations, I'm not sure which one is better in our case. @tsconn23 What do you think?
```suggestion
	query := `FOR a IN annotations FILTER a.layer == @app LET hosts = (a.host) RETURN DISTINCT hosts`
	bindVars := map[string]interface{}{
		"app": string(contracts.Application),
	}
	cursor, err := db.Query(ctx, query, bindVars)
```